### ISSUE TITLE
Custom cache set for Images uploaded

### ISSUE THREAD

**Issue by MCNorthwood** [PiranhaCMS/piranha.core#502]
Hi would it be possible to have a config setting similar to the page/post to be able to set the CacheControl a max-age for the images uploaded to blob storage?

The recommended amount of seconds set for static images should be 1 year (31536000 seconds)

Thanks

**Comment by tidyui** [comment]
Yes we could add this feature, however there’s no guarantee that uploaded files are THAT static as you can update the assets. With that kind of cache date you would need to disable media update and instead always upload new files.

As I see it the recommendation is for static assets not accessible for the editors such as logo, font files, stylesheets and so on. However it could be nice to actually mark a media asset as static, this would treat it in a special way and maybe set a longer expiration date on it.

Regards

**Comment by andi0b** [comment]
This feature is still missing, and it seems like Google is penalizing this behaviour.

The problem of updates is usually solved like this: You add a version number to the image `/images/location/image_800.jpg?v=11` and if you change the image again, the version number increases. So for the browser this is a new image then, and it gets downloaded again.

**Comment by tidyui** [comment]
This issue hasn’t been prioritized as you should be able to override the max-age of blob items by using azure cdn in front of it, but we’ll look into it!

**Comment by tidyui** [comment]
As for versioning images we don’t store versions for media items, but the last modification date could be used the generate a unique hash.

**Comment by vitali-karmanov** [comment]
I'm excited that this feature is being added to version 11. Looking forward to it, thank you!
